### ISSUE TITLE
A preserveOrder= of "false" does nothing

### ISSUE THREAD

**Issue by sydb** [TEIC/Stylesheets#241]
In an ODD, `<sequence preserveOrder="false">` produces the same RELAX NG as `<sequence>` (or `<sequence preserveOrder="true">`).

For testing, try [this tiny ODD](https://github.com/TEIC/Stylesheets/files/836590/Tpo.odd.txt) that tries to constraint `<choice>` to contain 1 `<expan>` and 1 `<abbr>` in either order. Then try it changing "false" to "true". (Note that, to appease GitHub, I have added a “.txt” extension to the filename.)

**Comment by sydb** [comment]
N.B. [TEI issue #1970](https://github.com/TEIC/TEI/issues/1970).

**Comment by martindholmes** [comment]
Pinging this ticket --  surely we need to get this fixed? It's currently not possible (as far as I can see) to create a content model which may contain zero or one of each of a list of elements in any order.

**Comment by joeytakeda** [comment]
I've recently been bitten by this bug and confirm it's still happening.

I've done a bit of digging in the Stylesheets and it looks like the root of the problem (or, at least, the initial problem) is in `odd2odd.xsl`. Running this command against the `dev` branch of the Stylesheets (where preserveOrder.odd is the ODD @sydb provided):

```
saxon -s:preserveOrder.odd -xsl:../Stylesheets/odds/odd2odd.xsl -o:preserveOrder_expanded.odd

```

Shows a `<sequence>` for content that has lost its `@preserveOrder` entirely

```
          <content>
            <sequence minOccurs="1" maxOccurs="1">
              <elementRef key="abbr"/>
              <elementRef key="expan"/>
            </sequence>
          </content>
```

I _think_ the problem is in the `odd2odd-simplifyODD` template and, in particular, this condition:

https://github.com/TEIC/Stylesheets/blob/96001864018d9708ab44523fcafe3475f6885d6f/odds/odd2odd.xsl#L1489-L1493

Only the `@minOccurs` and `@maxOccurs` is retained from the context element and all other attributes are ignored; just did a quick test by adding `@xml:id` to the `<sequence>` and it's also absent from `preserveOrder_expanded.odd`.

I'm happy to work on this in a fork and file a PR, if that would be helpful!

**Comment by joeytakeda** [comment]
The problem was indeed in `odd2odd.xsl` as far as I can tell. `teiodds.xsl` (imported by`odd2relax.xsl`) already has handling for `preserveOrder`:

https://github.com/TEIC/Stylesheets/blob/96001864018d9708ab44523fcafe3475f6885d6f/odds/teiodds.xsl#L2078-L2104


So copying over the attributes in `odd2odd.xsl` seems to fix the issue, preserving `@preserveOrder` so that it's recognized by the time the ODD hits the above template.

I've forked the repo and made the change to `odd2odd.xsl` (will do a PR once I post this comment) as well as added some tests to Test2. I wasn't sure if that was the proper way to add a test--I couldn't find any documentation on how to add tests to Test, but the documentation for Test2 was clear, so I just went ahead and did that; let me know if that's actually helpful and, if not, happy to remove the changes to Test2.
